platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/StopWatch.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.utility;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A simple stop watch that can be used to measure elapsed time.
+ */
+public class StopWatch {
+    private long startTime;
+    private long endTime;
+    private boolean running;
+    private boolean suspended;
+
+    /**
+     * Creates a new StopWatch.
+     */
+    public StopWatch() {
```

Comment:
I would prefer a second constructor that has our Time interface as param. The default constructor can use Time.getOsTime() for that. Instead of calling System.getNano() internally you get that from the Time instance. By doing so you can change the stopWatch behavior for tests.